.circleci/config.yml

```diff
@@ -366,17 +366,38 @@ jobs:
             bin/bootstrap.sh
             # See https://circleci.com/docs/2.0/testing-ios/#pre-starting-the-simulator
             xcrun instruments -w "iPhone 11 (13.2) [" || true
+      - run:
+          name: Build sample framework
+          command: |
+            # Build in Debug mode to speed it all up
+            pushd samples/ios/GleanSampleFramework
+            carthage bootstrap --platform iOS --no-use-binaries --cache-builds --configuration Debug --verbose
+            popd
+            bash bin/run-ios-sample-framework-build.sh
+      - run:
+          name: "Create Carthage bin-only project specification"
+          command: |
+            ZIP_URL=https://circleci.com/api/v1.1/project/github/mozilla/glean/$CIRCLE_BUILD_NUM/artifacts/0/dist/GleanSampleFramework.framework.zip
+            echo "{\"0.0.1\":\"$ZIP_URL\"}" > GleanSampleFramework.json
+      - store_artifacts:
+          path: GleanSampleFramework.framework.zip
+          destination: dist/GleanSampleFramework.framework.zip
+      - store_artifacts:
+          path: GleanSampleFramework.json
+          destination: dist/GleanSampleFramework.json
       - run:
           name: Use local checkout through Carthage
           command: |
             GLEAN_PATH="$(pwd)"
             GLEAN_COMMIT="${CIRCLE_SHA1}"
             CARTFILE_PATH="${GLEAN_PATH}/samples/ios/app/Cartfile"
+            ZIP_URL=https://circleci.com/api/v1.1/project/github/mozilla/glean/$CIRCLE_BUILD_NUM/artifacts/0/dist/GleanSampleFramework.framework.zip
```

Comment:
Aaaaah, so we're using the binary build of the sample library. That is ... neat.
Wonder if we should do that for Glean itself eventually.

Reply:
It did speed up the build, and I considered doing exactly what you said but I felt like that fell under the bug about speeding up the CI.

Reply:
Since we already saved the `Glean.framework.zip` to `/dist` in an earlier step, we should be able to use it for building both the Framework and the App, saving us from building Glean 3x (as Carthage runs for the framework and the app)